astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
+    @property
+    def tap(self):
+        """TAP service"""
+        if self._tap is None:
+            self._tap = pyvo.dal.TAPService(
+                f'{self.VO_URL}/tap', session=self._session
+            )
+            self._session = self._tap._session
+        return self._tap
+
+    @property
+    def _meta(self):
+        """Queries and holds meta-information about the tables.
+
+        This is a table that holds useful information such as
+        the list of default columns per table, the reasonable default
+        search radius per table that is appropriate for a mission etc.
+        Instead of making a server call for each of that type information,
+        we do a single one and then post-process the resulting table.
+
+        These are not meant to be used directly by the user.
+        """
+        if self._meta_info is None:
+            query = (
+                "SELECT split_part(name, '.', 1) AS table, "
+                "split_part(name, '.', 2) AS par, "
+                "CAST(value AS DECIMAL) AS value "
+                "FROM metainfo "
+                "WHERE (type = 'parameter' and relation = 'order') "
+                "OR relation LIKE 'defaultSearchRadius' "
+                "ORDER BY value"
+            )
+            self._meta_info = self.query_tap(query).to_table()
+            self._meta_info['value'] = np.array(
+                self._meta_info['value'], np.float32)
+            self._meta_info = self._meta_info[self._meta_info['value'] > 0]
+        return self._meta_info
+
+    def _get_default_cols(self, table_name):
+        """Get a list of default columns for a table
+
+        Parameters
+        ----------
+        table_name: str
+            The name of table as a str
+
+        Return
+        ------
+        a list of column names
 
-    def query_async(self, request_payload, *, cache=True, url=None):
         """
-        Submit a query based on a given request_payload. This allows detailed
-        control of the query to be submitted.
+        meta = self._meta[
+            (self._meta['table'] == table_name)
+            & (self._meta['par'] != '')
+        ]
+        meta.sort('value')
+        defaults = meta['par']
+        return defaults
+
+    def get_default_radius(self, table_name):
+        """Get a mission-appropriate default radius for a table
+
+        Parameters
+        ----------
+        table_name: str
+            The name of table as a str
+
+        Returns
+        -------
+        The radius as ~astropy.units
 
-        cache (bool) defaults to True. If set overrides global caching behavior.
-        See :ref:`caching documentation <astroquery_cache>`.
         """
+        meta = self._meta[
+            (self._meta['table'] == table_name)
+            & (self._meta['par'] == '')
+        ]
+        radius = np.double(meta['value'][0]) * u.arcmin
+        return radius
 
-        if url is None:
-            url = conf.server
+    def set_session(self, session):
```

Comment:
What's the purpose of having this method? The session should be set at inheritance and otherwise should not be in the user API

Reply:
same comment stands